### ISSUE TITLE
Suggest `Option::is_some_and(...)` for manual `matches!(x, Some(x) if ...)` implementations

### ISSUE THREAD

**Issue by Luk-ESC** [rust-lang/rust-clippy#13282]
### What it does

Title.
Should also work for Result and its variants (`is_ok_and`, `is_err_and`).

Found while going through warnings for [popsicle](https://github.com/pop-os/popsicle/), where clippy suggested to replace
```rs
let sensitive = match combo_box.active_text() {
    Some(text) if text.as_str() != "None" => true,
    _ => false,
};
```
with
```rs
let sensitive = matches!(combo_box.active_text(), Some(text) if text.as_str() != "None");
```
However, Clippy didn't go from `matches!` to the even more concise alternative using `is_some_and`:
```rs
let sensitive = combo_box.active_text().is_some_and(|text| text.as_str() != "None");
```

### Advantage

`is_variant_and` is shorter than the `matches!` alternative
`is_variant_and` results in code that is more readable

### Drawbacks

MSRV of 1.70.0

### Example

```rust
let is_empty = matches!(option, Some(x) if x.is_empty());
```

Could be written as:

```rust
let is_empty = option.is_some_and(|x| x.is_empty());
```
